src/librustc_yk_sections/emit_sir.rs

```diff
@@ -130,13 +130,14 @@ impl<'a, 'tcx> ConvCx<'a, 'tcx> {
         // Note that it is not necessary to adjust the successor block indices in the SIR
         // terminators due to this transformation. The user blocks will be pushed at new indices
         // which are already correct for the existing successor indices.
-        let skip = match self.tcx.sess.opts.cg.tracer {
-            TracerMode::Software => {
-                let num_mir_blks = self.mir.basic_blocks().len();
-                debug_assert!(num_mir_blks % 2 == 0);
-                num_mir_blks / 2
-            },
-            _ => 0,
+        let skip = if self.tcx.sess.opts.cg.tracer == TracerMode::Software &&
```

Comment:
This is a bit hard to read. Would it be better as:

```rust
let skip;
if ... {
   skip = ...;
}
else {
   skip = ...;
}
```

?

Reply:
I'm indifferent, so I'll update it :)